Scripts/DCS-BIOS/lib/modules/aircraft_modules/MiG-29A.lua

```diff
@@ -72,6 +78,71 @@ local devices = {
 	SO69 = 64,
 }
 
+local function cabinTempSwitchIntValue(arg_value)
+	local val = MiG_29A.round2(arg_value, 2)
+
+	if val < 0.075 then
+		return 0
+	elseif val < 0.30 then
+		return 1
+	elseif val < 0.60 then
+		return 2
+	else
+		return 3
+	end
+end
+
+function MiG_29A:defineCabinTempSwitch(identifier, device_id, arg_number, category, description)
+	local alloc = self:allocateInt(3)
+	self:addExportHook(function(dev0)
+		local val = cabinTempSwitchIntValue(dev0:get_argument_value(arg_number))
+		alloc:setValue(val)
+	end)
+
+	local control = Control:new(category, ControlType.toggle_switch, identifier, description, {
+		SetStateInput:new(3, "set the switch position"),
+		FixedStepInput:new("switch to previous or next state"),
+	}, {
+		IntegerOutput:new(alloc, Suffix.none, "switch position -- 0 = Middle, 1 = UP , 2 = LEFT, 3 = RIGHT"),
+	})
+	self:addControl(control)
+
+	self:addInputProcessor(identifier, function(toState)
+		local dev = GetDevice(device_id)
+
+		if dev == nil then
+			return
+		end
+
+		local currentState = cabinTempSwitchIntValue(GetDevice(0):get_argument_value(arg_number))
+		local new_state
+
+		if toState == "INC" and currentState == 3 then
+			return
+		elseif toState == "DEC" and currentState == 0 then
+			return
+		end
```

Comment:
I think these checks could be bundled into the checks below, so we're not checking if it's INC/DEC twice. like
```lua
if toState == "INC" then
    if currentState >= 3 then return end
    new_state = current_state + 1
    -- ...
end
```
or, just drop this at the end of that block
```lua
if new_state < 0 or new_state > 3 then return end
```

Reply:
Fixed!